src/modules/differential_drive_control/DifferentialDriveKinematics/DifferentialDriveKinematics.cpp

```diff
@@ -0,0 +1,68 @@
+#include "DifferentialDriveKinematics.hpp"
+
+
+void DifferentialDriveKinematics::setInput(const matrix::Vector2f &input, bool inverse)
+{
+	_input = input;
+
+	if (inverse) {
+		_linear_vel_x = _input(0);
+		_yaw_rate = _input(1);
+		computeInverseKinematics();
+
+	} else {
+		_motor_vel_right = _input(0);
+		_motor_vel_left = _input(1);
+		computeForwardsKinematics();
+	}
+
+}
+
+void DifferentialDriveKinematics::computeInverseKinematics()
```

Comment:
Couldn't you make it simpler by removing the internal states from this class and define all the inputs as function arguments? Like
`computeInverseKinematics(float linear_vel_x, float yaw_rate, float wheel_base, float wheel_radius)`?

Reply:
Yes this is a simpler approach, will implement.